src/traversing.js

```diff
@@ -168,10 +159,18 @@ jQuery.each({
 		return elem.contentDocument || jQuery.merge( [], elem.childNodes );
 	}
 }, function( name, fn ) {
+	var // Sorting (which implicitly de-dupes) is required for non-adjacent traversals
+		sort = name !== "prev" && name !== "next",
```

Comment:
I think we all upset about this, https://github.com/jquery/jquery/pull/180 was a big win, right know, i'm not totally sure that https://github.com/jquery/jquery/issues/1766 is necessary, owner of this ticket found it by looking at the code, i'm not sure if it's real-life problem, whereas optimization was landed four years ago and know one notice the issue, but everybody notice the speed improvement.

Maybe we just document and forget about it?